src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationToken.java

```diff
@@ -79,6 +79,8 @@ public Object getCredentials() {
 
     @Override
     public Object getPrincipal() {
+    	if (this.userDetails != null)
+    		return this.userDetails;
```

Comment:
Not sure it's a big deal, but the indentation on this line seems to be too much on the right...